### PR TITLE
HPCC-13781 Prevent large row arrays from looping forever when resizing

### PR DESCRIPTION
**Pull request by ghalliday**
If the nunber of rows added to a DynamicRoxieOutputRowArray was close to
2^32, the code for calculating the next allocation size could loop
indefinitely (or for a very long time).

Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@jakesmith please review

File: roxie/roxiemem/roxierowbuff.cpp [hunk 1]
```diff
@@ -150,7 +150,14 @@ bool DynamicRoxieOutputRowArray::ensure(rowidx_t requiredRows)
         //   reduce fragmentation.
         //Use 25% for the moment.  It should possibly be configurable - e.g., higher for thor global sort.
         while (newSize < requiredRows)
-            newSize += newSize/4;
+        {
+            rowidx_t nextSize = newSize + newSize / 4;
+            //check to see if it has wrapped
+            if (nextSize < newSize)
+                newSize = requiredRows;
```
Comment (jakesmith) [inline roxie/roxiemem/roxierowbuff.cpp]: worth adding a 'break;' here?

**Comment by jakesmith** [comment]
@ghalliday - 1 minor comment.

**Comment by ghalliday** [comment]
I added the break - likely to be a marginal effect, but probably clearer.

@richardkchapman please review.

**Pull request merged**